src/commands/react-native/upload.ts

```diff
@@ -132,7 +133,6 @@ export class UploadCommand extends Command {
       this.config,
       {
         apiKey: process.env.DATADOG_API_KEY,
-        datadogSite: process.env.DATADOG_SITE,
```

Comment:
This creates a small regression in the case where the site is both specified in the environment and the `datadog-ci.json` file with different values.

Before the order of priority used to be: env > config file > default value ('datadoghq.com')
Now since the site from env is put in the default value it will just be: config file  > env.

It's only a tiny case but that also introduces a different logic for api key and site variables, which can make debugging harder.

I would suggest the following change here (we can either keep or remove the change above):
```suggestion
        datadogSite: process.env.DATADOG_SITE || process.env.DD_SITE,
```